react/FilePicker/index.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react'
+import { makeStyles } from '@material-ui/core/styles'
+import PropTypes from 'prop-types'
+
+import { FixedDialog } from '../CozyDialogs'
+import FilePickerHeader from './FilePickerHeader'
+import FilePickerFooter from './FilePickerFooter'
+import FilePickerBody from './FilePickerBody'
+
+const getIsAccept = accept => {
+  // If accept is falsy, set default to file
+  const acceptSelection = accept
+    ? accept.replaceAll(' ', '').split(',')
+    : ['file']
+
+  return {
+    file: acceptSelection.includes('file'),
+    folder: acceptSelection.includes('folder')
+  }
+}
+
+const useStyles = makeStyles(theme => ({
+  paper: {
+    height: '100%',
+    '& >div:nth-child(4)': {
+      padding: 0
+    },
+    [theme.breakpoints.down('sm')]: {
+      '& >div:first-child': {
+        display: 'none'
+      }
+    }
+  }
+}))
+
+const ROOT_DIR_ID = 'io.cozy.files.root-dir'
+
+const FilePicker = ({ onClose, onChange, accept, multiple }) => {
+  const classes = useStyles()
+  const [folderId, setFolderId] = useState(ROOT_DIR_ID)
+  const [filesIdsSelected, setFilesIdsSelected] = useState([])
+
+  const selectFileId = fileId => {
+    if (!multiple) {
+      setFilesIdsSelected([fileId])
+    } else {
+      setFilesIdsSelected(fileId)
+    }
+  }
+
+  const navigateTo = folder => setFolderId(folder.id)
+  const handleConfirm = () => {
+    onChange(filesIdsSelected)
+    onClose()
+  }
+  const isAccept = getIsAccept(accept)
+
+  return (
+    <FixedDialog
+      open
+      disableTitleAutoPadding
+      onClose={onClose}
+      size="large"
+      classes={{
+        paper: classes.paper
+      }}
+      title={<FilePickerHeader navigateTo={navigateTo} folderId={folderId} />}
+      content={
+        <FilePickerBody
+          navigateTo={navigateTo}
+          selectFileId={selectFileId}
+          filesIdsSelected={filesIdsSelected}
+          folderId={folderId}
+          isAccept={isAccept}
```

Comment:
As `isAccept` is only used by `FilePickerBody` I would move `getIsAccept` inside of it.

Also I find this name hard to understand. I expected to find a boolean, but it is a dictionary of booleans. So I'm wondering if there is a better name (I don't have any propositon sorry)

Reply:
I will rename it `fileTypesAccepted`